msauth/resource_owner_password_grant.go

```diff
@@ -0,0 +1,27 @@
+package msauth
+
+import (
+	"context"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/microsoft"
+)
+
+// ResourceOwnerPasswordGrant preforms OAuth 2.0 client resource owner password grant and returns a token.
+func (m *Manager) ResourceOwnerPasswordGrant(ctx context.Context, tenantID, clientID, clientSecret, username, password string, scopes []string) (oauth2.TokenSource, error) {
+	endpoint := microsoft.AzureADEndpoint(tenantID)
+	endpoint.AuthStyle = oauth2.AuthStyleInParams
+	config := &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Endpoint:     endpoint,
+		Scopes:       scopes,
+	}
+	var err error
+	t, err := config.PasswordCredentialsToken(ctx, username, password)
+	ts := config.TokenSource(ctx, t)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	t, err := config.PasswordCredentialsToken(ctx, username, password)
	if err != nil {
		return nil, err
	}
	ts := config.TokenSource(ctx, t)
```

- `var err error` is unneeded.
- It should return immediately if it gets an error.  `t` might cause nil pointer dereference.

Reply:
Thanks for the feedback. I've made the changes and updated the README.

Reply:
Looks great.  Thank you!